src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -3321,13 +3321,31 @@ Definition of bitvector addition as a nat.
 @[simp] theorem add_ofFin (x : BitVec n) (y : Fin (2^n)) :
   x + .ofFin y = .ofFin (x.toFin + y) := rfl
 
+@[simp]
```

Comment:
I am not sure you want to mark those lemmas `simp`, as this transformations looses some semantic information that may sometimes be desirable.

Reply:
what information?

Reply:
By automatically simplifying to `toNat` we move away from the bitvec world into the `Nat` world. One consequence of this is that `bv_decide` won't understand this statement anymore. I did not realize this when I wrote the comment above, but losing the ability to use `bv_decide` is nothing we want to do by default.